### PR TITLE
Add maxEncodeSize configuration for zstd compression

### PR DESCRIPTION
**Pull request by glorrian**
As mentioned in #11343, Zstd encoder in Netty has a unique parameter of maxEncoded size, and if users of Micronaut want to use Zstd with files more than ≈31Mb Netty will throw an exception like "requested encode buffer size exceeds the maximum allowable size) so I thing we should enable change this parameters for MK's users if they would like to handle bigger files at MK server

### CODE REVIEW

**Comment by glorrian** [comment]
@yawkat @graemerocher @dstepanov I hope that I did everything right and I will be glad to receive a review. I can't write tests because its need to add a library https://github.com/luben/zstd-jni and i can't do it, because it's not in netty-bom and I can't connect it transitively because it's declared with optional=true in netty poms, I'm not sure about its license, so I didn't add it to libs.versions.toml

File: http-server-netty/src/main/java/io/micronaut/http/server/netty/HttpCompressionStrategy.java [hunk 1]
```diff
@@ -39,4 +39,9 @@ public interface HttpCompressionStrategy extends Toggleable {
     default int getCompressionLevel() {
         return StandardCompressionOptions.gzip().compressionLevel();
     }
+
+    /**
+     * @return The maximum size of data that can be encoded using the zstd algorithm.
+     */
+    int getMaxZstdEncodeSize();
```
Comment (yawkat) [inline http-server-netty/src/main/java/io/micronaut/http/server/netty/HttpCompressionStrategy.java:46]: Please add a default implementation for compatibility

**Review by yawkat — APPROVED** [review]

File: http-server-netty/src/main/java/io/micronaut/http/server/netty/HttpCompressionStrategy.java (hunk 1 above)
Comment (glorrian) [reply http-server-netty/src/main/java/io/micronaut/http/server/netty/HttpCompressionStrategy.java:46]: it already there

**Review by sdelamo — COMMENTED** [review]
Could we add a test?

**Comment by glorrian** [comment]
> Could we add a test?

I'm not sure about this dependency, but formally we can 
> I can't write tests because its need to add a library https://github.com/luben/zstd-jni and i can't do it, because it's not in netty-bom and I can't connect it transitively because it's declared with optional=true in netty poms, I'm not sure about its license, so I didn't add it to libs.versions.toml

**Comment by glorrian** [comment]
Also i've tested it manually and it works on my local but i didn't run any auto tests

**Comment by glorrian** [comment]
@sdelamo

**Pull request merged**